### PR TITLE
Add imports to terraform plan result

### LINKED ISSUE #4401: Support Terraform version v1.5.0 and later
**What would you like to be added**:

The import feature is added from Terraform version v1.5.0 breaks our Terraform logics. It had been fixed by PR https://github.com/pipe-cd/pipecd/pull/4396, but we should also update the diff detection logic for Terraform, since import should be counted as a trigger for new deployment as well.

**Why is this needed**:

### PR DESCRIPTION
**Pull request by khanhtc1202**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #4401

**Does this PR introduce a user-facing change?**:

```release-note
Support Terraform v1.5.0 and later which introduces import feature
```

### CODE REVIEW

File: pkg/app/piped/platformprovider/terraform/terraform.go [hunk 1]
```diff
@@ -329,7 +330,33 @@ func stripAnsiCodes(str string) string {
...
 
 func parsePlanResult(out string, ansiIncluded bool) (PlanResult, error) {
-	parseNums := func(add, change, destroy string) (adds int, changes int, destroys int, err error) {
+	parseNums := func(vals ...string) (imports, adds, changes, destroys int, err error) {
+		if len(vals) < 3 || len(vals) > 4 {
+			err = fmt.Errorf("invalid plan result: %v", vals)
+			return
+		}
+
+		var impt, add, change, destroy string
```
[the remark this answers is not in the document]
Comment (khanhtc1202) [reply pkg/app/piped/platformprovider/terraform/terraform.go:339]: Take a close look, you can see these variables are not returned one, but they are used to store vals[i] data 😄

[the remark this answers is not in the document]
Comment (sivchari) [reply pkg/app/piped/platformprovider/terraform/terraform.go:339]: Oops, I overlooked 🐱

File: pkg/app/piped/platformprovider/terraform/terraform_test.go [hunk 2]
```diff
@@ -48,3 +48,54 @@ func TestPlanHasChangeRegex(t *testing.T) {
...
+
+	for _, tc := range testcases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+			result, err := parsePlanResult(tc.input, false)
+			assert.Equal(t, tc.expectedErr, err != nil)
+			if err != nil {
+				assert.Equal(t, tc.expected, result)
+			}
```
Comment (kentakozuka) [inline pkg/app/piped/platformprovider/terraform/terraform_test.go:98]: This test passes even if the `expected` is unexpected.
How about simply writing this?

```suggestion
			assert.Equal(t, tc.expected, result)
```

**Review by kentakozuka — CHANGES_REQUESTED** [review]
Nice :)
Left some comments. PTAL

File: pkg/app/piped/platformprovider/terraform/terraform.go [hunk 3]
```diff
@@ -349,13 +376,14 @@ func parsePlanResult(out string, ansiIncluded bool) (PlanResult, error) {
 		out = stripAnsiCodes(out)
 	}
 
-	if s := planHasChangeRegex.FindStringSubmatch(out); len(s) == 4 {
-		adds, changes, destroys, err := parseNums(s[1], s[2], s[3])
+	if s := planHasChangeRegex.FindStringSubmatch(out); len(s) > 0 {
```
[the remark this answers is not in the document]
Comment (khanhtc1202) [reply pkg/app/piped/platformprovider/terraform/terraform.go:379]: Oops, you're right. Have to fix the regex to extract both import available and not available case 👍

[the remark this answers is not in the document]
Comment (khanhtc1202) [reply pkg/app/piped/platformprovider/terraform/terraform.go]: Updated, thank you 👍

**Comment by khanhtc1202** [comment]
@kentakozuka PTAL when you have time 🙏

**Review by kentakozuka — APPROVED** [review]
Superb 😄

**Pull request merged**